src/legacy.rs

```diff
@@ -0,0 +1,194 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use crate::{
+    aesgcm::{AesGcmEceWebPush, AesGcmEncryptedBlock},
+    common::WebPushParams,
+    crypto::EcKeyComponents,
+    error::*,
+};
+
+/// Encrypt a block using legacy AESGCM encoding.
+///
+/// param remote_pub &[u8] - The remote public key
+/// param remote_auth &u8 - The remote authorization token
+/// param salt &[u8] - The locally generated random salt
+/// param data &[u8] - The data to encrypt
+///
+pub fn encrypt_aesgcm(
+    remote_pub: &[u8],
+    remote_auth: &[u8],
+    salt: &[u8],
+    data: &[u8],
+) -> Result<AesGcmEncryptedBlock> {
+    let cryptographer = crate::crypto::holder::get_cryptographer();
+    let remote_key = cryptographer.import_public_key(remote_pub)?;
+    let local_key_pair = cryptographer.generate_ephemeral_keypair()?;
+    let mut padr = [0u8; 2];
+    cryptographer.random_bytes(&mut padr)?;
+    // since it's a sampled random, endian doesn't really matter.
+    let pad = ((usize::from(padr[0]) + (usize::from(padr[1]) << 8)) % 4095) + 1;
+    let params = WebPushParams::new(4096, pad, Vec::from(salt));
+    AesGcmEceWebPush::encrypt_with_keys(&*local_key_pair, &*remote_key, &remote_auth, data, params)
+}
+
+/// Decrypt a block using legacy AESGCM encoding.
+///
+/// param components &str - The locally generated private key components.
```

Comment:
The `components` argument should be documented as an `&EcKeyComponents`, not a `&str`. (I notice that the existing `decrypt` function has the same problem, unfortunately; given that rustdoc generates really nice automated function signature references, maybe we should just remove this narrative "param foo &bar" documentation entirely?)

Reply:
That's right, I copied and pasted the comment but did not pay attention to the problem. I'll add a fixing commit.